grafana/grafana
# InfluxDB: Fix querying with hardcoded retention policy

**What is this feature?**

We weren't using InfluxDB retention policy and we hardcoded the retention policy as `default` and opt-out `default` named retention policy while querying.  With [this PR](https://github.com/grafana/grafana/pull/63820) we added retention policy support and removed hardcoded retention policy. That PR wasn't covering the existing dashboard panels with hardcoded policies. So that causes datasource is trying to query with non-existing retention policies. Since only a handful of users are using the retention policy in the queries that was slipped through. 

This PR fixes querying non-existing hardcoded retention policies and uses the default retention policy. This is a seamless operation. **Users will still see hardcoded retention policy in the editor dropdown unless they explicitly choose the retention policy they want to use.** 

#### What should I do?
- Go and select the retention policy explicitly and save the panel/dashboard
Check the following images to see the hardcoded retention policy

Hardcoded Policy:
![image](https://user-images.githubusercontent.com/820946/231815506-073752ab-1131-4a82-87ea-480f99b25c62.png)

Actual Retention Policies:
![image](https://user-images.githubusercontent.com/820946/231815550-41ec4aa2-3057-4795-9e3d-ba1d812d0734.png)


**Why do we need this feature?**

To support users who have hardcoded retention policy in their panels.

**Who is this feature for?**

InfluxDB Users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/support-escalations/issues/5540

**Special notes for your reviewer:**

#### How to test it?
- Set up an influxdb `influxql` dashboard with minimal query
  - Make sure you do not have a retention policy named `default`
  - `gdev-influxdb1-influxql` datasource would work fine
  - Run `make devenv sources=influxdb1` to have that datasoruce running
- Go to dashboard json model and change the target policy as `"policy": "default"` and save it
  - `default` was the hardcoded policy
- Go back to the dashboard 
- See panel query is not working
- Switch to this branch
- Refresh the page and see panel query is working and there is no error

Please check that:


# Release notice breaking change
`default` named retention policies won't be used to query. Users who have a `default` named retention policy in their `influxdb` database, have to rename it to something else. Having `default` named retention policy is not breaking anything. We will make sure to use the actual default retention policy under the hood. To change the hardcoded retention policy in the `dashboard.json`, users must they select the right retention policy from dropdown and save the panel/dashboard.
